schema/categories_schema.cue

```diff
@@ -0,0 +1,41 @@
+package shopify
+
+#categoryId: =~"^gid://shopify/Taxonomy/Category/\\w+(-\\d+)*$"
+
+// Lookup to ensure all attributes exist
+_attribute_lookup: {
+	for k in attributes {
+		"\(k.id)": k.name
+	}
+}
+
+// Structure for attribute references
+#attribute_ref: {
+	id:   =~"^gid://shopify/Taxonomy/Attribute/\\d+$"
+	name: _attribute_lookup[id]
+}
+
+// Structure for category references
+#category_ref: {
+	id:   string & #categoryId
+	name: string
+}
+
+// Structure for categories, including recursive definitions for children, attributes, and ancestors
+#category: {
+	id:        string & #categoryId
+	name:      string
+	level:     int & >=0
+	full_name: string
+	parent_id: *null | string & #categoryId
+	attributes: [...#attribute_ref]
+	children: [...#category_ref] | *[]
```

Comment:
Not sure the `| *[]` is necessary as we always have a `children`/`ancestors` key. Validation passes locally without it.